Nos/Models/NosNotification+CoreDataClass.swift

```diff
@@ -16,16 +16,21 @@ public class NosNotification: NSManagedObject {
 
     class func createIfNecessary(
         from eventID: HexadecimalString, 
-        authorKey: HexadecimalString, 
+        date: Date?,
+        authorKey: HexadecimalString,
         in context: NSManagedObjectContext
     ) throws -> NosNotification? {
+        guard let date, date > cutoffDate() else {
```

Comment:
Since we need a `date` to proceed, does it make sense to make the parameter non-optional?